### PR TITLE
[SPARK-44298][BUILD] Disable PySpark test on the daily test of Java 21 before the new arrow version release

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to disable PySpark test on the daily test of Java 21 before the new arrow version release.

### Why are the changes needed?
Make daily testing runs based on Java21 successful.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ on:
       java:
         required: false
         type: string
-        default: 8
+        default: 21-ea
```
Comment (panbingkun) [inline .github/workflows/build_and_test.yml:28]: Only for test.

**Comment by panbingkun** [comment]
1.Before
https://github.com/apache/spark/actions/runs/5440417950/jobs/9893341039
image

2.After
https://github.com/panbingkun/spark/actions/runs/5444102515/jobs/9914235099
image

File: dev/sparktestsupport/modules.py [hunk 2]
```diff
@@ -436,75 +436,85 @@ def __hash__(self):
     ],
 )
```
Comment (panbingkun) [inline dev/sparktestsupport/modules.py:438]: I will only use `pyspark_sql` as a test here. If this idea is reasonable, I will proceed with the other modules.

Comment (LuciferYang) [reply dev/sparktestsupport/modules.py:438]: cc @HyukjinKwon @dongjoon-hyun FYI

Comment (HyukjinKwon) [reply dev/sparktestsupport/modules.py:438]: It makes sense to skip when there are few but I think we shouldn't just skip them like this if there are too many.

Comment (HyukjinKwon) [reply dev/sparktestsupport/modules.py:438]: We should fix instead of just skipping them.

Comment (LuciferYang) [reply dev/sparktestsupport/modules.py:438]: Or for the daily test of  Java 21, disable PySpark test first? The failed tests also need wait for the new arrow version be upgrade.

Comment (HyukjinKwon) [reply dev/sparktestsupport/modules.py:438]: Yeah, we could just don't run PySpark tests for now in the daily build.

Comment (panbingkun) [reply dev/sparktestsupport/modules.py:438]: Ok，let me to do it.

Comment (panbingkun) [reply dev/sparktestsupport/modules.py]: Done

**Review by LuciferYang — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by LuciferYang** [comment]
Merged to master. Thanks @panbingkun and @HyukjinKwon

**Pull request closed without merging**